src/main/java/org/folio/consortia/repository/SharingInstanceRepository.java

```diff
@@ -1,12 +1,47 @@
 package org.folio.consortia.repository;
 
+import static java.util.Objects.nonNull;
+
+import java.util.ArrayList;
 import java.util.UUID;
 
 import org.folio.consortia.domain.entity.SharingInstanceEntity;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface SharingInstanceRepository extends JpaRepository<SharingInstanceEntity, UUID> {
+public interface SharingInstanceRepository
+  extends JpaRepository<SharingInstanceEntity, UUID>, JpaSpecificationExecutor<SharingInstanceEntity> {
+
+  interface Specifications {
+    static Specification<SharingInstanceEntity> constructSpecification(UUID instanceIdentifier, String sourceTenantId,
+        String targetTenantId, String status) {
+      var list = new ArrayList<Specification<SharingInstanceEntity>>();
+      if(nonNull(instanceIdentifier)) list.add(byInstanceIdentifier(instanceIdentifier));
+      if(nonNull(sourceTenantId)) list.add(bySourceTenantId(sourceTenantId));
+      if(nonNull(targetTenantId)) list.add(byTargetTenantId(targetTenantId));
```

Comment:
Lets use StringUtils.isNotEmpty() to avoid searches by empty string

Reply:
Fixed.